File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        
        if not headA or not headB:
            return None

        currentA = headA
        currentB = headB

        while currentA != currentB:
            currentA = currentA.next if currentA else headB
            currentB = currentB.next if currentB else headA

        return currentA
            
        

        



        

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode* pointerA = headA;
        ListNode* pointerB = headB;

        while(pointerA != pointerB) {
            if(pointerA == NULL) {
                pointerA = headB;
            }
            else {
                pointerA = pointerA->next;
            }
            if(pointerB == NULL) {
                pointerB = headA;
                
            }
            else {
                pointerB = pointerB->next;
            }


        }
        return pointerB;
        
    }
};